sql/src/main/java/io/crate/analyze/CreateViewAnalyzer.java

```diff
@@ -25,30 +25,47 @@
 import io.crate.analyze.relations.QueriedRelation;
 import io.crate.analyze.relations.RelationAnalyzer;
 import io.crate.expression.symbol.Field;
+import io.crate.expression.symbol.format.SymbolPrinter;
+import io.crate.metadata.Functions;
 import io.crate.metadata.TableIdent;
 import io.crate.metadata.TransactionContext;
+import io.crate.sql.parser.SqlParser;
 import io.crate.sql.tree.CreateView;
+import io.crate.sql.tree.Query;
 
 public final class CreateViewAnalyzer {
 
     private final RelationAnalyzer relationAnalyzer;
+    private final SQLPrinter sqlPrinter;
 
-    CreateViewAnalyzer(RelationAnalyzer relationAnalyzer) {
+    CreateViewAnalyzer(Functions functions, RelationAnalyzer relationAnalyzer) {
+        this.sqlPrinter = new SQLPrinter(new SymbolPrinter(functions));
         this.relationAnalyzer = relationAnalyzer;
     }
 
     public AnalyzedStatement analyze(CreateView createView, TransactionContext txnCtx, String defaultSchema) {
         TableIdent name = TableIdent.of(createView.name(), defaultSchema);
+        name.validate();
         QueriedRelation query = (QueriedRelation) relationAnalyzer.analyzeUnbound(
             createView.query(), txnCtx, ParamTypeHints.EMPTY);
 
+        // sqlPrinter isn't feature complete yet; so restrict CREATE VIEW to only support queries where the
+        // format->analyze round-trip works.
+        String formattedQuery;
+        try {
+            formattedQuery = sqlPrinter.format(query);
+            relationAnalyzer.analyzeUnbound((Query) SqlParser.createStatement(formattedQuery), txnCtx, ParamTypeHints.EMPTY);
+        } catch (Exception e) {
+            throw new UnsupportedOperationException("Query cannot be used in a VIEW: " + createView.query());
```

Comment:
This would also report the query being unsupported if it contained syntax errors. Would be nice to give a reason for it being not supported, e.g. pass the caught exception as a cause.

Reply:
In that case the initial analysis should have failed: https://github.com/crate/crate/pull/7097/files#diff-9ae66d466ae636d48a99ad8aa96e211dR49

Reply:
Ah yes, perfect.